### PR TITLE
Adjoh/merge master to dev

### PR DESCRIPTION
**Pull request by AdamBJohnsonx**
Merge master (3.4.5) into dev.

Testing-wise, I locally ran the tests in common, and then I ran the instrumented tests in common.

Implies MSAL: https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/1464
Implies Broker: https://github.com/AzureAD/ad-accounts-for-android/pull/1627
Implied ADA: https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/1628

### CODE REVIEW

File: azure-pipelines/templates/steps/credscan-policheck.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# File: credscan-policheck.yml
```
Comment (AdamBJohnsonx) [inline azure-pipelines/templates/steps/credscan-policheck.yml:1]: This file is defunct.

File: common/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -36,4 +40,4 @@
         </intent>
     </queries>
 
-</manifest>
+</manifest>
```
Comment (shahzaibj) [inline common/src/main/AndroidManifest.xml:43]: nit: new line ending

File: common/src/main/java/com/microsoft/identity/common/internal/configuration/LibraryConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+     * Controls whether interactive authorization activities (Browser, Embedded, Broker) are
+     * launched in the task associated with the activity provided as a parameter to interactive requests
+     * The current default behavior of common is to launch the activity in a new Task.
+     * This creates effectively 2 task stacks (which can appear as 2 windows in multi-window configurations)
+     * The 2 task stacks allows for unexpected user experience when navigating away for authorization UI
+     * when the authorizaiton is still in process.
+     */
+    private boolean authorizationInCurrentTask;
+
+}
```
Comment (shahzaibj) [inline common/src/main/java/com/microsoft/identity/common/internal/configuration/LibraryConfiguration.java:76]: nit: new line ending

File: testutils/src/main/java/com/microsoft/identity/internal/testutils/TestUtils.java [hunk 4]
```diff
@@ -72,12 +75,8 @@ public static SharedPreferencesFileManager getSharedPreferences(final String sha
...
     public static SharedPreferencesFileManager getEncryptedSharedPreferences(final String sharedPrefName) {
         final Context context = ApplicationProvider.getApplicationContext();
-        final SharedPreferencesFileManager barePreferences = SharedPreferencesFileManager.getSharedPreferences(
-                context,
-                sharedPrefName,
-                Context.MODE_PRIVATE,
-                new AndroidCommonComponents(context).
-                        getStorageEncryptionManager(null));
+        final StorageHelper storageHelper = new StorageHelper(context);
+        final SharedPreferencesFileManager barePreferences = SharedPreferencesFileManager.getSharedPreferences(context, sharedPrefName, storageHelper);
```
Comment (shahzaibj) [inline testutils/src/main/java/com/microsoft/identity/internal/testutils/TestUtils.java:79]: I think we need the code that was there on dev

File: testutils/build.gradle [hunk 5]
```diff
@@ -198,6 +198,7 @@ dependencies {
         implementation(group: 'com.microsoft.identity', name: 'labapi', version: '0.0.+')
     }
 
+    implementation "org.mockito:mockito-inline:$rootProject.ext.mockitoCoreVersion"
```
Comment (shahzaibj) [inline testutils/build.gradle:201]: Once we address this: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1472/files#r671562680 then probably don't need this depedency

**Review by shahzaibj — CHANGES_REQUESTED** [review]
Left a comment about using the code that there was on dev for TestUtils (which is the one I think we need with the refactoring)

File: common/src/main/java/com/microsoft/identity/common/internal/configuration/LibraryConfiguration.java (hunk 3 above)
Comment (AdamBJohnsonx) [reply common/src/main/java/com/microsoft/identity/common/internal/configuration/LibraryConfiguration.java:76]:
```suggestion
}

```

File: common/src/main/AndroidManifest.xml (hunk 2 above)
Comment (AdamBJohnsonx) [reply common/src/main/AndroidManifest.xml:43]:
```suggestion
</manifest>

```

File: common/src/main/java/com/microsoft/identity/common/internal/ui/AuthorizationStrategyFactory.java [hunk 6]
```diff
@@ -145,4 +153,4 @@ private AuthorizationAgent validAuthorizationAgent(final AuthorizationAgent agen
             return agent;
         }
     }
-}
+}
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/ui/AuthorizationStrategyFactory.java:156]:
```suggestion
}

```

File: testutils/src/main/java/com/microsoft/identity/internal/testutils/TestUtils.java (hunk 4 above)
Comment (AdamBJohnsonx) [reply testutils/src/main/java/com/microsoft/identity/internal/testutils/TestUtils.java:79]: half of it.

Comment (AdamBJohnsonx) [reply testutils/src/main/java/com/microsoft/identity/internal/testutils/TestUtils.java:79]:
```suggestion
       final SharedPreferencesFileManager barePreferences = SharedPreferencesFileManager.getSharedPreferences(
           context,
           sharedPrefName,
           new AndroidCommonComponents(context).
                             getStorageEncryptionManager(null))
      );
```

File: testutils/build.gradle (hunk 5 above)
Comment (shahzaibj) [reply testutils/build.gradle:201]: Do we need this Mockito depedency? Not clear if this is being used somewhere in this PR

**Review by shahzaibj — APPROVED** [review]

File: testutils/build.gradle (hunk 5 above)
Comment (AdamBJohnsonx) [reply testutils/build.gradle:201]:
```suggestion
```

**Review by shoatman — APPROVED** [review]
Thanks Adam.

File: CODEOWNERS [hunk 7]
```diff
@@ -5,78 +5,78 @@
 # Repo Code Owners
 #--------------------
 
-*       @AdamBJohnsonx @iambmelt @rpdome @shahzaibj @shoatman
+*       @iambmelt @rpdome @shahzaibj @AzureAD/androidleads
```
Comment (AdamBJohnsonx) [inline CODEOWNERS:8]:
```suggestion
*       @AdamBJohnsonx @iambmelt @rpdome @shahzaibj @shoatman @AzureAD/androidleads
```

File: CODEOWNERS [hunk 8]
```diff
@@ -5,78 +5,78 @@
...
 
 
 # Area Owners
 #---------------------
 
 # Interfaces
-/common4j/src/main/java/com/microsoft/identity/common/java/interfaces/              @rpdome @iambmelt @AdamBJohnsonx
+/common4j/src/main/java/com/microsoft/identity/common/java/interfaces/              @rpdome @iambmelt @AzureAD/androidleads
```
Comment (AdamBJohnsonx) [inline CODEOWNERS:15]:
```suggestion
/common4j/src/main/java/com/microsoft/identity/common/java/interfaces/              @rpdome @iambmelt @AdamBJohnsonx @AzureAD/androidleads
```

File: CODEOWNERS [hunk 9]
```diff
@@ -5,78 +5,78 @@
...
-/common4j/src/test/com/microsoft/identity/common/java/net                           @rpdome @AdamBJohnsonx
-/common4j/src/test/com/microsoft/identity/common/java/platform                      @shahzaibj @rpdome @AdamBJohnsonx
-/common4j/src/test/com/microsoft/identity/common/java/providers                     @rpdome @iambmelt @shoatman
-/common4j/src/test/com/microsoft/identity/common/java/telemetry                     @shahzaibj @rpdome @iambmelt
-/common4j/src/test/com/microsoft/identity/common/java/util                          @rpdome @iambmelt @AdamBJohnsonx
+/common/src/test/                                                                   @shahzaibj @rpdome @iambmelt @AzureAD/androidleads
+/common/src/androidTest/                                                            @shahzaibj @rpdome @shoatman @iambmelt @AzureAD/androidleads 
+/common4j/src/test/                                                                 @shahzaibj @rpdome @shoatman @iambmelt @AzureAD/androidleads
+/uiautomationutilities/src/                                                         @shahzaibj @iamgusain @AzureAD/androidleads
```
Comment (AdamBJohnsonx) [inline CODEOWNERS:21]:
```suggestion
/common/src/test/                                                                   @shahzaibj @rpdome @iambmelt @AdamBJohnsonx @AzureAD/androidleads
/common/src/androidTest/                                                            @shahzaibj @rpdome @shoatman @iambmelt @AzureAD/androidleads 
/common4j/src/test/                                                                 @shahzaibj @rpdome @shoatman @iambmelt @AdamBJohnsonx @AzureAD/androidleads
/uiautomationutilities/src/                                                         @AdamBJohnsonx @shahzaibj @iamgusain @AzureAD/androidleads
```

**Pull request merged**